### PR TITLE
pkg/trace/info: Remove BytesEstimated

### PR DESCRIPTION
**Pull request by LotharSee**
### What does this PR do?

Remove Trace Writer BytesEstimated and its reported timeseries.
BytesEstimated was never meant to be relevant for customers nor our support and was never used. It was only helpful for our development.
As such, I consider this doesn't require a CHANGELOG entry.

### Motivation

Less confusion about what our Trace Agent self-monitoring report (one less timeseries, less code to explain).

### CODE REVIEW

File: pkg/trace/info/writer.go [hunk 1]
```diff
@@ -25,7 +25,6 @@ type TraceWriterInfo struct {
 	Retries           atomic.Int64
 	Bytes             atomic.Int64
 	BytesUncompressed atomic.Int64
```
Comment (gbbr) [inline pkg/trace/info/writer.go:27]: `BytesUncompressed` was only there to compare against `BytesEstimated` and be able to tell the accuracy of it. Do you think it's still useful going forward? The only thing we could gain from it is to analyse the efficiency of the compression. If not, maybe we can remove that too.

**Review by gbbr — APPROVED** [review]
LGTM! Just one q.

File: pkg/trace/info/writer.go (hunk 1 above)
Comment (LotharSee) [reply pkg/trace/info/writer.go:27]: I see value in this metric both to monitor compression efficiency (more especially as we'll work on improving it soon) and as a way to estimate uncompressed ingestion (which is what's measured for span ingestion billing) per host.

**Review by jtappa — APPROVED** [review]
👍🏻 from docs

**Pull request merged**